### DISCUSSION TITLE
add another map

### DISCUSSION THREAD

**Discussion by NitroDenko097** [kus/cs2-modded-server#113] [Q&A] [answered]
how add another map on my server?

**Comment by mavproductions** [comment]
Hi @NitroDenko097,

You can add a map by editing the subscribed IDs and gamemodes_server.txt file in the root csgo directory. 

Just add to the map group you want it to belong to and restart your server and you should be good from there!

**Reply by NitroDenko097** [reply]
it doesn't work, because I entered the id, from the game cmd it works but when I try to go to the minigames mode then I try to find it in the rtv it gives me de_dust 2 and other maps that no longer fit with the minigames

**Comment by NitroDenko097** [comment]
sorry, 1 more question, how do I solve this
![unknown_2024 05 26-21 34](https://github.com/kus/cs2-modded-server/assets/170464306/25bd8a62-9172-4224-b00c-5b1927bc1f94)

**Reply by mavproductions** [reply]
Just `!rcon bot_kick` will do

**Reply by mavproductions** [reply]
You need to also fix any JSON errors you might have in your counterstrikesharp/config/admin.json file

**Reply by NitroDenko097** [reply]
Can I contact you somewhere privately to send all the settings?

**Reply by NitroDenko097** [reply]
because I put it everywhere but it still doesn't work

**Reply by mavproductions** [reply]
> Can I contact you somewhere privately to send all the settings?
> 

Sure add me a steam https://steamcommunity.com/id/mavproductions/

**Reply by Vichoko** [reply]
This is because you need to define both admin.json and admin_groups.json for that plugin to work.

**Reply by kus** [reply]
You need to update [admin_groups.json](https://github.com/kus/cs2-modded-server/blob/master/custom_files_example/addons/counterstrikesharp/configs/admin_groups.json) and [admins.json](https://github.com/kus/cs2-modded-server/blob/master/custom_files_example/addons/counterstrikesharp/configs/admins.json) (these are what I use on my server) and both files must be valid JSON which you can check here https://jsonlint.com/

**Reply by GooberOTF** [reply]
> > Can I contact you somewhere privately to send all the settings?
> 
> Sure add me a steam https://steamcommunity.com/id/mavproductions/

Hey Mav! please add me on steam. i would love to ask you some questions and try out your server you got going using this repo :) I am more than willing to help you out for your time :) 
https://steamcommunity.com/profiles/76561199578946575

Thanks!

**Comment by NitroDenko097** [comment]
I entered my api key and registered in all admin,json but it doesn't work

**Reply by mavproductions** [reply]
Are you able to verify your API key is loaded into the server successfully?

Edit: if you type `sv_setsteamaccount` into your terminal for the server, it should reply back with `Warning: game server already logged into steam. You need to use the sv_setsteamaccount command earlier.`

**Comment by Vichoko** [comment, marked as answer]
To add a new map do what I did on https://github.com/kus/cs2-modded-server/pull/89.
You basically need to add the map into two files.

**Reply by kus** [reply]
@NitroDenko097 This is the correct answer, further to this answer; you can edit the files on your server where they will be overwritten when you pull updates from my repo unless you handle merges manually.

Or you can copy the files to your [custom_files](https://github.com/kus/cs2-modded-server?tab=readme-ov-file#custom-files) directory and make the changes there, but the only disadvantage of that is you won't get any of the repo updates for the files you add there. But you can always copy them across manually.

I have [instructions here](https://github.com/kus/cs2-modded-server?tab=readme-ov-file#setting-maps-for-different-game-modes) on how to add maps into your own `custom_files` folder.

The whole reason the `custom_files` folder exists in this repo is so you can use the repo as a "base" for your server then add your modifications their and always stay up to date with the repo.

The [custom_files_example](https://github.com/kus/cs2-modded-server/tree/master/custom_files_example) folder is what I personally use on my server. It sets the servername and the admins which is a good to go off if you are just starting. You can delete the `custom_files` folder and rename `custom_files_example` to `custom_files` and make your changes in there.
